File: Recursion/kth_symbol_in_grammer.py

def grammer(n,k):
    if (n==1):

        
        return 0;
        # /* code */
    
    mid=int(pow(2,n-1)/2)
    if (k<=mid):
    
        return grammer(n-1,k);
        # /* code */
    
    else:
    
        rem=k%mid;
        return 1-grammer(n-1,rem)
        # /* code */
print(grammer(4,5))

File: Recursion/kth_symbol_in_grammer.cpp
#include<bits/stdc++.h>
using namespace std;
int grammer(int n,int k){
    if (n==1)
    {

        
        return 0;
        /* code */
    }
    int mid=pow(2,n-1)/2;
    if (k<=mid)
    {
        return grammer(n-1,k);
        /* code */
    }
    else
    {
        int rem=k%mid;
        return 1-grammer(n-1,rem);
        /* code */
    }
    
    
    
}
int main(){
    cout<<grammer(4,5);
}